internal/controllers/cryostat_controller_test.go

```diff
@@ -337,7 +337,83 @@ var _ = Describe("CryostatController", func() {
 
 				Expect(sa.ImagePullSecrets).To(Equal(oldSA.ImagePullSecrets))
 				Expect(sa.Secrets).To(Equal(oldSA.Secrets))
-				Expect(sa.AutomountServiceAccountToken).To(BeNil())
+				Expect(sa.AutomountServiceAccountToken).To(Equal(oldSA.AutomountServiceAccountToken))
+			})
+		})
+		Context("with an existing Role", func() {
+			var cr *operatorv1beta1.Cryostat
+			var oldRole *rbacv1.Role
+			BeforeEach(func() {
+				cr = test.NewCryostat()
+				oldRole = test.OtherRole()
+				t.objs = append(t.objs, cr, oldRole)
+			})
+			It("should update the Role", func() {
+				t.reconcileCryostatFully()
+
+				role := &rbacv1.Role{}
+				err := t.Client.Get(context.Background(), types.NamespacedName{Name: "cryostat", Namespace: "default"}, role)
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(metav1.IsControlledBy(role, cr)).To(BeTrue())
+
+				// Labels are unaffected
+				Expect(role.Labels).To(Equal(oldRole.Labels))
+
+				// Rules should be fully replaced
+				Expect(role.Rules).To(Equal(test.NewRole().Rules))
+			})
+		})
+		Context("with an existing Role Binding", func() {
+			var cr *operatorv1beta1.Cryostat
+			var oldBinding *rbacv1.RoleBinding
+			BeforeEach(func() {
+				cr = test.NewCryostat()
+				oldBinding = test.OtherRoleBinding()
+				t.objs = append(t.objs, cr, oldBinding)
+			})
+			It("should update the Role Binding", func() {
+				t.reconcileCryostatFully()
+
+				binding := &rbacv1.RoleBinding{}
+				err := t.Client.Get(context.Background(), types.NamespacedName{Name: "cryostat", Namespace: "default"}, binding)
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(metav1.IsControlledBy(binding, cr)).To(BeTrue())
+
+				// Labels are unaffected
+				Expect(binding.Labels).To(Equal(binding.Labels))
```

Comment:
I think the second one should be `oldBinding`?

Reply:
Good catch! This would have tested nothing otherwise.